lib/ordnance_survey/api/v1/search_places.rb

```diff
@@ -0,0 +1,66 @@
+module OrdnanceSurvey
+  class Api
+    class V1
+      class SearchPlaces
+        def initialize(client:)
+          self.client = client
+        end
+
+        def index(params:)
+          mapped_params = {
+            postcode: params[:postcode].delete(" ")
+          }
+
+          response = client.get(path: "/search/places/v1/postcode", params: mapped_params)
+
+          return nil unless response
+
+          results = []
+
+          response[:results].each do |result|
+            results << {
+              address: result[:DPA][:ADDRESS],
+              address_line_1: address_line_1(sub_building_name: result[:DPA][:SUB_BUILDING_NAME], building_name: result[:DPA][:BUILDING_NAME]),
+              address_line_2: result[:DPA][:THOROUGHFARE_NAME],
+              address_line_4: result[:DPA][:POST_TOWN],
+              postcode: result[:DPA][:POSTCODE]
+            }
+          end
+
+          results
+        end
+
+        def show(params:)
+          mapped_params = {
+            query: [params[:address_line_1], params[:postcode].delete(" ")].join(", "),
+            maxresults: 1
+          }
+
+          response = client.get(path: "/search/places/v1/find", params: mapped_params)
```

Comment:
@nomasprime There are two separate endpoints according to the Ordnance Survey API documentation. The ones in the original ticket from @mashakir were both the same, but it turns out if we only want a Postcode we should use the specific endpoint for that. Hence having two separate endpoints

Reply:
Feels like the endpoint should probably be the same, one with and one without `maxresults`. Is this not the case?

Technically this class is now dealing with two resources, not critical but worth noting.

Reply:
@nomasprime I explained before the original spec in the ticket is wrong. We were told we could use the same endpoint but it returns inaccurate results due to how we intend to use it - building number and postcode. So the `find` endpoint is still being used for that.
However when we supply just a postcode for greater accuracy we should be using I believe the `postcode` endpoint.
Documentation is attached if you want to review. Happy to discuss further.
[Ordnance Survey Places Technical Specification](https://osdatahub.os.uk/docs/places/technicalSpecification)

There are two separate methods. Both use the same URL except the last slug. Are you saying for `find` and `postcode` they should be in separate classes? I did ask this a few days ago, and my understanding - which I might have gotten wrong - was that it would be ok